plumbing/cst/chain_state.go

```diff
@@ -171,3 +190,29 @@ func (chn *ChainStateReadWriter) SetHead(ctx context.Context, key types.TipSetKe
 	}
 	return chn.readWriter.SetHead(ctx, headTs)
 }
+
+// ChainExport exports nodes current chain to `out`
+func (chn *ChainStateReadWriter) ChainExport(ctx context.Context, head types.TipSetKey, out io.Writer) (types.TipSetKey, error) {
+	headTS, err := chn.GetTipSet(head)
+	if err != nil {
+		return types.UndefTipSet.Key(), err
+	}
+	logStore.Info("starting CAR file export")
+	headKey, err := chain.Export(ctx, headTS, chn.readWriter, chn.messageProvider, out)
```

Comment:
Maybe I misled with an earlier comment - there's no need to return the head key if it's already passed as an argument.